net-irc/bitchx/patches/bitchx-1.2.1.patchset

```diff
@@ -0,0 +1,25 @@
+From 4d89af6a3dbb5c25243c1a752a0284dfc9fe3579 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Zolt=C3=A1n=20Mizsei?= <zmizsei@extrowerk.com>
+Date: Thu, 8 Nov 2018 20:41:07 +0100
+Subject: HAIKU: Build fix
+
+
+diff --git a/include/irc_std.h b/include/irc_std.h
+index 29400c2..b6c2dff 100644
+--- a/include/irc_std.h
++++ b/include/irc_std.h
+@@ -205,9 +205,11 @@ int	killpg (int pgrp, int sig);
+ char *	getpass (const char * prompt);
+ #endif
+ 
++#ifndef __HAIKU__
+ #ifndef BCOPY_DECLARED
+ void	bcopy (const void *src, void *dest, size_t n);
```

Comment:
Which header declares bcopy() on Haiku? The configure script looks for it in <string.h>, ideally we'd fix this in configure.

Reply:
[`bcopy()`](http://xref.plausible.coop/source/xref/haiku/headers/posix/strings.h#24) in Haiku is a macro, not a function.

Reply:
Ahh I see.  In that case you could change

```
-#ifndef BCOPY_DECLARED
+#if !defined(BCOPY_DECLARED) && !defined(bcopy)

```
The best fix upstream is to replace all remaining uses of `bcopy()` with `memmove()`.